src/mesh/parallel/fci.cxx

```diff
@@ -53,8 +53,9 @@ inline BoutReal sgn(BoutReal val) { return (BoutReal(0) < val) - (val < BoutReal
 
 // Calculate all the coefficients needed for the spline interpolation
 // dir MUST be either +1 or -1
-FCIMap::FCIMap(Mesh &mesh, int dir, bool zperiodic)
-  : dir(dir), boundary_mask(mesh), corner_boundary_mask(mesh), y_prime(&mesh) {
+FCIMap::FCIMap(Mesh &mesh_in, int dir, bool zperiodic)
+  : mesh(mesh_in), dir(dir), boundary_mask(mesh), corner_boundary_mask(mesh),
```

Comment:
We use `mesh_in` to set `mesh` and then use `mesh` to set things like the `boundary_mask`. Whilst this should work ok it's sensitive to order -- would it be safer to use `mesh_in` everywhere here? @ZedThree may have some comments one way or the other.

Reply:
The variables are always initialised in the order they are declared in the class, so there is already an order dependency there. There should be a warning generated if this is wrong -- if there's no warning, it should be safe!

Note that this will conflict with #1345, so might be best to wait for that to go in then revisit this.

Reply:
In that case I think I'd prefer the use of `mesh_in` everywhere here -- it's easy for someone to change the order of variable declarations in the separate header without realising they're being used like this.